racket/src/expander/extract/simplify-defn.rkt

```diff
@@ -20,20 +21,90 @@
       (seteq)
       (set-union (car args) (apply union-all (cdr args)))))
 
-;; compute the variables that are the target of a set! in e
+; This function gets called when encountering a 'suspect letrec': some
+; letrec expression where the bound variable is used somewhere inside
+; the bound expression, and we can't proove (given the information
+; available here) that uses is safe: that the implicit mutation that
+; happens as the letrec defines the variables.
+;
+; Generally, we want to emit a warning in this case, as it's possible
+; that we stumbled on an actual bug (e.g. the bug fixed in git commit
+; b9b017261bf35cc42b341ca5d3b8f17e955416c2.) However, there are also
```

Comment:
Since this commit is stable (in master), it might be nicer to use "reference formatting" (what `git show -s --format=reference` does): `b9b017261b (io: repair wrong-variable use in checking for special Windows paths, 2025-06-03)`.